### PR TITLE
feat: auditlogs design refactor cw1764

### PR DESCRIPTION
**Pull request by vishnu-narayanan**
# Pull Request Template

## Description

Changelog

- [x] The table should take the full width.
- [x] The column with the header Activity should take the 60% of the width of the screen.
- [x] Change the design to 3 columns - Activity, Date, IP Address
- [x] Refactor the log text to account for the switch to `Activity` column

Fixes https://linear.app/chatwoot/issue/CW-1764/update-the-design

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Tested locally

image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/settings/auditlogs/Index.vue [hunk 1]
```diff
@@ -109,4 +141,11 @@ export default {
   word-break: break-all;
   white-space: normal;
 }
+.full-width-table {
+  width: 100%;
+}
+
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/auditlogs/Index.vue:147]:
```suggestion

```

File: app/javascript/dashboard/routes/dashboard/settings/auditlogs/Index.vue [hunk 2]
```diff
@@ -16,8 +16,13 @@
 
         <table
           v-if="!uiFlags.fetchingList && records.length"
-          class="woot-table"
+          class="woot-table full-width-table"
         >
+          <colgroup>
+            <col style="width: 60%" />
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/auditlogs/Index.vue:22]:
```suggestion
            <col class="column-activity" />
```

**Review by nithindavid — CHANGES_REQUESTED** [review]
Please see the comments

File: app/javascript/dashboard/routes/dashboard/settings/auditlogs/Index.vue [hunk 3]
```diff
@@ -88,6 +92,34 @@ export default {
...
+    generateLogText(auditLogItem) {
+      const username =
+        auditLogItem.username !== null
+          ? auditLogItem.username
+          : this.$t('AUDIT_LOGS.ACTION.SYSTEM');
+      const auditableType = auditLogItem.auditable_type.toLowerCase();
+      const action = auditLogItem.action.toLowerCase();
+
+      if (auditLogItem.action === 'create') {
+        return `${username} ${this.$t(
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/auditlogs/Index.vue:104]: As per the spec, the name should have the preference. Resolve the name from the agent model in the vuex. If it is not available, then use the email as the fallback.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Please see the comment about the name.

File: app/javascript/dashboard/routes/dashboard/settings/auditlogs/Index.vue (hunk 3 above)
Comment (vishnu-narayanan) [reply app/javascript/dashboard/routes/dashboard/settings/auditlogs/Index.vue]: done

**Comment by vishnu-narayanan** [comment]
image

File: app/javascript/dashboard/routes/dashboard/settings/auditlogs/Index.vue [hunk 4]
```diff
@@ -1,8 +1,8 @@
 <template>
   <div class="column content-box">
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/auditlogs/Index.vue:2]: @vishnu-narayanan Is it possible to relocate the footer to the bottom of the screen? Currently, it appears to be shifting whenever we navigate to a different page, causing a jarring experience.

```suggestion
  <div class="column content-box audit-log--settings">
```

Add this to the bottom as style tag

```
<style lang="scss" scoped>
.audit-log--settings {
  display: flex;
  justify-content: space-between;
  flex-direction: column;

  .remote-address {
    width: 14rem;
  }

  .wrap-break-words {
    word-break: break-all;
    white-space: normal;
  }
  .column-activity {
    width: 60%;
  }
}
</style>
```

**Review by iamsivin — COMMENTED** [review]
@vishnu-narayanan I have included some comments in the PR. Could you take a look?

**Comment by vishnu-narayanan** [comment]
@iamsivin Requested changes have been made. Could you please take a look and approve?

**Review by iamsivin — APPROVED** [review]
Tested LGTM 💯

**Pull request merged**